File: aoc_2023/day_1/trebuchet.py
import re
from utils.file import read_file


def is_digit(str: str) -> bool:
    return regex.match(str) is not None


contents = read_file(__file__, "input")
lines = contents.split("\n")
sum = 0
pat = r"^\d$"
regex = re.compile(pat)
for line in lines:
    if not line:
        continue
    chars: str = "".join(filter(is_digit, line))
    num = int(chars[0] + chars[-1])
    sum += num


print(sum)

File: aoc_2023/day_1/trebuchet.rs
#[path = "../../utils/file.rs"]
mod file;
use regex::Regex;

fn main() {
    let contents = file::read_file(file_abs!());
    let lines = contents.split("\n").filter(|v| !v.is_empty());
    let re = Regex::new(r"^\d$").unwrap();
    let mut sum = 0;
    for line in lines {
        let chars: Vec<_> = line.split("").filter(|char| re.is_match(char)).collect();
        let num = (chars[0].to_owned() + chars.last().unwrap())
            .parse::<i32>()
            .unwrap();
        sum += num
    }
    println!("Result: {}", sum)
}
